### PR TITLE
[GLOSSARY] Glossary additions

### PR DESCRIPTION
**Pull request by Chalarangelo**
## Description

In continuation of the development of #690.

List of terms (13):

- Variable
- Constant
- NPM
- Yarn
- Express
- React
- Git
- Repository
- Argument
- Function
- ES6
- IP
- HTML

## What does your PR belong to?
- [ ] Website
- [ ] Snippets
- [x] General / Things regarding the repository (like CI Integration)
- [ ] Tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking improvement of a snippet)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have checked that the changes are working properly
- [x] I have checked that there isn't any PR doing the same
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: glossary/Argument.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+### Argument
+
+An argument is a value passed as an input to a function and can be either a primitve or an object.
```
Comment (flxwu) [inline glossary/Argument.md]: You may also add here that in JS, you can pass a function as function argument as well as functions are first-class objects in JavaScript

File: glossary/Function.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+### Function
+
+Functions are self-contained blocks of code with their own scope, that can be called by other code and are usually associated with a unique identifier.
+Functions accept input in the form of arguments and can optionally return an output. 
+JavaScript functions are also objects.
```
Comment (flxwu) [inline glossary/Function.md]: could amend it here as well

File: glossary/Git.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+### Git
+
+Git is an open-source version control system, used for source code management.
+Git allows users to clone and edit repositories on their local machines, without having to change the master repository.
```
Comment (flxwu) [inline glossary/Git.md]: this is pretty vague... I would reword it 😅  also, I would change the last sentence to "the remote master repository"

File: glossary/React.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+### React
+
+React is a frontend framework, that allows developers to create component-based user interfaces.
```
Comment (flxwu) [inline glossary/React.md]: would add "dynamic" here as keyword

File: glossary/Repository.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+### Repository
+
+In a version control system, a repository (or repo for short) is a data structure that stores metadata for a set of files.
```
Comment (flxwu) [inline glossary/Repository.md]: i.e. a project? 😅

File: glossary/Yarn.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+### Yarn
+
+Yarn is a package manager for the JavaScript programming language.
```
Comment (flxwu) [inline glossary/Yarn.md]: maybe add "by facebook" here?

File: glossary/Function.md (hunk 2 above)
Comment (Chalarangelo) [reply glossary/Function.md]: For this term, I think stating that they're object is sufficient..

File: glossary/Argument.md (hunk 1 above)
Comment (Chalarangelo) [reply glossary/Argument.md]: Definitely worth stating here, I'll update the term.

File: glossary/Git.md (hunk 3 above)
Comment (Chalarangelo) [reply glossary/Git.md]: Honestly, it's based on Wikipedia and other sources. It's the least vague I could write. Could you just reword it on the same branch? Maybe you can explain it better than that.

File: glossary/React.md (hunk 4 above)
Comment (Chalarangelo) [reply glossary/React.md]: Will be done.

File: glossary/Repository.md (hunk 5 above)
Comment (Chalarangelo) [reply glossary/Repository.md]: Yup.

File: glossary/Yarn.md (hunk 6 above)
Comment (Chalarangelo) [reply glossary/Yarn.md]: Honestly, I prefer not to say who is behind what as packages have a slighttendency to change ownership over time and it might get outdated or sound like unnecessary promo. I'm thinking we can link to some external content from the glossary like said packages etc. so it will be obvious to people what is what.

**Comment by Chalarangelo** [comment]
@flxwu Chanes made as requested and comments added for the terms I did not update.

File: glossary/Function.md (hunk 2 above)
Comment (skatcat31) [reply glossary/Function.md]: > Functions are special objects in JavaScript that have their own local scope and can be called by other pieces of code. They are usually assigned a unique identifier, but sometimes an anonymous function is used. Functions can optionally accept inputs and return outputs.

Combines both but may be confusing. It should also be noted that a Function call will always return. If one is not explicit, the implicit return of `undefined` is returned.

File: glossary/Argument.md (hunk 1 above)
Comment (skatcat31) [reply glossary/Argument.md]: I'd put it as a note like
> ... or an object(a function is an object)

or some other way

Comment (Chalarangelo) [reply glossary/Argument.md]: Hmmm, I could rephrase the first sentence to be something along the lines of *An argument is a value passed as an input to a function and can be either a primitve, a function or an object.* Seems less weird that way.

File: glossary/Function.md (hunk 2 above)
Comment (Chalarangelo) [reply glossary/Function.md]: I prefer the phrasing asis. The return functionality, however, should be explained. I'll update.

File: glossary/Git.md (hunk 3 above)
Comment (Chalarangelo) [reply glossary/Git.md]: Updated. Tried to keep the terminology present as it could be of use to beginners.

File: glossary/Yarn.md (hunk 6 above)
Comment (Chalarangelo) [reply glossary/Yarn.md]: Currently it only works with JS. Plus, this is a JS-specific repo, so we can quietly omit that information at least until this is a stable feature.

**Comment by Chalarangelo** [comment]
@skatcat31 Made some updates, read my comments for the rest.

**Comment by PaulReiber** [comment]
primitve

**Comment by Chalarangelo** [comment]
@PaulReiber fixed, thanks for the heads up!

File: glossary/Yarn.md (hunk 6 above)
Comment (skatcat31) [reply glossary/Yarn.md]: In that case it might be better as

> Yarn is a package manager made by Facebook. It can be used as an alternative to the npm package manager and is compatible with the public NPM registry.

reason the second NPM is capitalized: It's actually the company name, no the tool

**Comment by Chalarangelo** [comment]
@skatcat31 @flxwu I think most terms are ready to go, what do you guys think?
